lib/node_modules/@stdlib/repl/lib/syntax_highlighter.js

```diff
@@ -125,6 +142,148 @@ setNonEnumerableReadOnly( SyntaxHighlighter.prototype, '_highlightLine', functio
+	this._theme = theme;
+});
+
+/**
+* Gets a theme's colors palette.
+*
+* @name getThemeConfig
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @param {string} theme - theme name
+* @throws {Error} must provide an existing theme name
+* @returns {Object} theme object
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'getThemeConfig', function getThemeConfig( theme ) {
+	if ( !contains( objectKeys( this._themes ), theme ) ) {
+		throw new Error( format( 'invalid argument. First argument must be an existing theme name. Value: `%s`.', theme ) );
+	}
+	return this._themes[ theme ];
+});
+
+/**
+* Sets the default color theme.
+*
+* @name setDefaultTheme
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @param {string} theme - theme name
+* @throws {Error} argument must be an existing theme
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'setDefaultTheme', function setDefaultTheme( theme ) {
+	if ( !contains( objectKeys( this._themes ), theme ) ) {
+		throw new Error( format( 'invalid argument. First argument must be an existing theme name. Value: `%s`.', theme ) );
+	}
+	this._defaultTheme = theme;
+});
+
+/**
+* Adds a new color theme.
+*
+* @name addTheme
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @param {Object} theme - theme object
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'addTheme', function addTheme( name, theme ) {
+	this._themes[ name ] = theme;
+});
+
+/**
+* Renames a color theme.
+*
+* @name renameTheme
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @param {string} oldName - old theme name
+* @param {string} newName - new theme name
+* @throws {Error} first argument must be an existing theme name
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'renameTheme', function renameTheme( oldName, newName ) {
+	if ( !contains( objectKeys( this._themes ), oldName ) ) {
+		throw new Error( format( 'invalid argument. First argument must be an existing theme name. Value: `%s`.', oldName ) );
+	}
+	this._themes[ newName ] = this._themes[ oldName ];
+	this._themes = omit( this._themes, oldName );
+});
+
+/**
+* Deletes a color theme.
+*
+* @name deleteTheme
+* @memberof SyntaxHighlighter.prototype
+* @type {Function}
+* @param {Object} theme - theme name
+* @throws {Error} must provide an existing theme name
+* @throws {Error} must provide a theme other than the default theme
+*/
+setNonEnumerableReadOnly( SyntaxHighlighter.prototype, 'deleteTheme', function deleteTheme( theme ) {
+	if ( !contains( objectKeys( this._themes ), theme ) ) {
+		throw new Error( format( 'invalid argument. First argument must be an existing theme name. Value: `%s`.', theme ) );
+	}
+	if ( theme === this._defaultTheme ) {
+		throw new Error( format( 'invalid argument. First argument cannot be the default theme. Value: `%s`.', theme ) );
+	}
+	if ( theme === this._theme ) {
+		this._highlightedLine = '';
+		this._theme = this._defaultTheme;
```

Comment:
I see that this is a case where `defaultTheme` is useful. However, I think it would be fine to not make `defaultTheme` something public. Instead, we'd reset the theme to some built-in default (e.g., `stdlib-ansi-basic`), without the user being able to specify otherwise.